### PR TITLE
damldocs: Show generic template args and context.

### PR DESCRIPTION
**Pull request by associahedron**
This PR collects the generic template args and context and displays it in the Rst and Markdown docs. The only thing left to finish generic template support in damldocs (for now) is collecting and displaying template instances.

### CODE REVIEW

File: compiler/damlc/tests/daml-test-files/ProposalDSL.EXPECTED.rst [hunk 1]
```diff
@@ -10,7 +10,7 @@ Templates
 
 .. _template-proposaldsl-proposal-62786:
 
-template **Proposal**
+template (Template t) => **Proposal** t
```
Comment (hurryabit) [inline compiler/damlc/tests/daml-test-files/ProposalDSL.EXPECTED.rst]: Tough question: Do we need the parentheses around `Template t`? :D

File: compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs [hunk 2]
```diff
@@ -88,7 +88,11 @@ renderSimpleRst ModuleDoc{..} = mconcat $
 tmpl2rst :: TemplateDoc -> RenderOut
 tmpl2rst TemplateDoc{..} = mconcat $
   [ renderAnchor td_anchor
-  , renderLine $ "template " <> enclosedIn "**" (unTypename td_name)
+  , renderLineDep $ \env -> T.concat
+      [ "template "
+      , maybe "" ((<> " => ") . type2rst env) td_super
+      , T.unwords (enclosedIn "**" (unTypename td_name) : td_args)
```
[the remark this answers is not in the document]
Comment (associahedron) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: That's a bold suggestion :-)

[the remark this answers is not in the document]
Comment (hurryabit) [reply compiler/damlc/daml-doc/src/DA/Daml/Doc/Render/Rst.hs]: :medal_sports: from the _Head of Puns_ for this response.

File: compiler/damlc/tests/daml-test-files/ProposalDSL.EXPECTED.rst (hunk 1 above)
Comment (associahedron) [reply compiler/damlc/tests/daml-test-files/ProposalDSL.EXPECTED.rst]: It's an artifact of how the contexts are stored and displayed, but at least it's very consistent accross damldocs. I'll fix it at some point.

**Pull request merged**